src/EditorFeatures/Core/Shared/Preview/PreviewSolutionCrawlerRegistrationService.cs

```diff
@@ -68,6 +74,34 @@ public void Unregister(Workspace workspace, bool blockingShutdown = false)
             {
                 source.Cancel();
             }
+
+            ClearDiagnostics(workspace);
+
+            AssertCleanup();
+        }
+
+        private void ClearDiagnostics(Workspace workspace)
+        {
+            var analyzer = _provider.CreateIncrementalAnalyzer(workspace);
+            var args = _service.GetDiagnosticsUpdatedEventArgs(workspace, projectId: null, documentId: null, cancellationToken: CancellationToken.None);
+
+            foreach (var documentId in args.Select(a => a.DocumentId).Distinct())
+            {
+                analyzer.RemoveDocument(documentId);
```

Comment:
> if those analyzer reported anything, then those should be removed when unregistered

How could the analyzer report anything?  It was literally created 5 lines prior.  When would it have reported anything?